### ISSUE TITLE
Random entity rendering crashes

### ISSUE THREAD

**Issue by wozniak** [IrisShaders/Iris#2391]
### Minecraft Version

Minecraft 1.21

### Iris Version

iris-1.7.2+mc1.21.jar

### Sodium Version

sodium-fabric-0.5.9+mc1.21.jar

### Operating System

Linux

### What is your GPU?

AMD Radeon RX 6600

### Java Version

openjdk 22

### Reproduction Steps

- Load Sildur's shaders (haven't tried with others, unsure)
- Play the game until it crashes

### Crash Report file and latest.log

```shell
---- Minecraft Crash Report ----
// This doesn't make any sense!

Time: 2024-06-29 22:56:41
Description: Rendering entity in world

java.lang.IllegalStateException: Missing elements in vertex: Normal
	at net.minecraft.class_287.method_60806(class_287.java:127)
	at net.minecraft.class_287.method_60805(class_287.java:96)
	at net.minecraft.class_287.method_22912(class_287.java:156)
	at net.minecraft.class_4588.method_22918(class_4588.java:532)
	at net.minecraft.class_382.method_22944(class_382.java:50)
	at net.minecraft.class_327$class_5232.method_27531(class_327.java:282)
	at net.minecraft.class_327.method_27530(class_327.java:299)
	at net.minecraft.class_327.method_1723(class_327.java:151)
	at net.minecraft.class_327.method_22942(class_327.java:84)
	at net.minecraft.class_327.method_30882(class_327.java:80)
	at net.minecraft.class_897.method_3926(class_897.java:209)
	at net.minecraft.class_1007.method_4213(class_1007.java:176)
	at net.minecraft.class_1007.method_3926(class_1007.java:45)
	at net.minecraft.class_897.method_3936(class_897.java:96)
	at net.minecraft.class_922.method_4054(class_922.java:155)
	at net.minecraft.class_1007.method_4215(class_1007.java:68)
	at net.minecraft.class_1007.method_3936(class_1007.java:45)
	at net.minecraft.class_898.method_3954(class_898.java:148)
	at net.minecraft.class_761.method_22977(class_761.java:1312)
	at net.minecraft.class_761.invokeRenderEntity(class_761.java)
	at net.irisshaders.iris.shadows.ShadowRenderer.renderEntities(ShadowRenderer.java:636)
	at net.irisshaders.iris.shadows.ShadowRenderer.renderShadows(ShadowRenderer.java:505)
	at net.irisshaders.iris.pipeline.IrisRenderingPipeline.renderShadows(IrisRenderingPipeline.java:1020)
	at net.minecraft.class_761.handler$zcg000$iris$renderTerrainShadows(class_761.java:5777)
	at net.minecraft.class_761.method_22710(class_761.java:977)
	at net.minecraft.class_757.method_3188(class_757.java:1087)
	at net.minecraft.class_757.method_3192(class_757.java:850)
	at net.minecraft.class_310.method_1523(class_310.java:1285)
	at net.minecraft.class_310.method_1514(class_310.java:882)
	at net.minecraft.client.main.Main.main(Main.java:256)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:243)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:278)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.minecraft.class_287.method_60806(class_287.java:127)
	at net.minecraft.class_287.method_60805(class_287.java:96)
	at net.minecraft.class_287.method_22912(class_287.java:156)
	at net.minecraft.class_4588.method_22918(class_4588.java:532)
	at net.minecraft.class_382.method_22944(class_382.java:50)
	at net.minecraft.class_327$class_5232.method_27531(class_327.java:282)
	at net.minecraft.class_327.method_27530(class_327.java:299)
	at net.minecraft.class_327.method_1723(class_327.java:151)
	at net.minecraft.class_327.method_22942(class_327.java:84)
	at net.minecraft.class_327.method_30882(class_327.java:80)
	at net.minecraft.class_897.method_3926(class_897.java:209)
	at net.minecraft.class_1007.method_4213(class_1007.java:176)
	at net.minecraft.class_1007.method_3926(class_1007.java:45)
	at net.minecraft.class_897.method_3936(class_897.java:96)
	at net.minecraft.class_922.method_4054(class_922.java:155)
	at net.minecraft.class_1007.method_4215(class_1007.java:68)
	at net.minecraft.class_1007.method_3936(class_1007.java:45)
	at net.minecraft.class_898.method_3954(class_898.java:148)
	at net.minecraft.class_761.method_22977(class_761.java:1312)
	at net.minecraft.class_761.invokeRenderEntity(class_761.java)
	at net.irisshaders.iris.shadows.ShadowRenderer.renderEntities(ShadowRenderer.java:636)
	at net.irisshaders.iris.shadows.ShadowRenderer.renderShadows(ShadowRenderer.java:505)
	at net.irisshaders.iris.pipeline.IrisRenderingPipeline.renderShadows(IrisRenderingPipeline.java:1020)
	at net.minecraft.class_761.handler$zcg000$iris$renderTerrainShadows(class_761.java:5777)
	at net.minecraft.class_761.method_22710(class_761.java:977)
	at net.minecraft.class_757.method_3188(class_757.java:1087)

-- Entity being rendered --
Details:
	Entity Type: minecraft:player (net.minecraft.class_745)
	Entity ID: 263168
	Entity Name: OnSpray
	Entity's Exact location: 625.65, 96.00, 748.90
	Entity's Block location: World: (625,96,748), Section: (at 1,0,12 in 39,6,46; chunk contains blocks 624,-64,736 to 639,319,751), Region: (1,1; contains chunks 32,32 to 63,63, blocks 512,-64,512 to 1023,319,1023)
	Entity's Momentum: 0.00, 0.00, 0.00
	Entity's Passengers: []
	Entity's Vehicle: null
Stacktrace:
	at net.minecraft.class_898.method_3954(class_898.java:148)
	at net.minecraft.class_761.method_22977(class_761.java:1312)
	at net.minecraft.class_761.invokeRenderEntity(class_761.java)
	at net.irisshaders.iris.shadows.ShadowRenderer.renderEntities(ShadowRenderer.java:636)
	at net.irisshaders.iris.shadows.ShadowRenderer.renderShadows(ShadowRenderer.java:505)
	at net.irisshaders.iris.pipeline.IrisRenderingPipeline.renderShadows(IrisRenderingPipeline.java:1020)
	at net.minecraft.class_761.handler$zcg000$iris$renderTerrainShadows(class_761.java:5777)
	at net.minecraft.class_761.method_22710(class_761.java:977)
	at net.minecraft.class_757.method_3188(class_757.java:1087)
	at net.minecraft.class_757.method_3192(class_757.java:850)
	at net.minecraft.class_310.method_1523(class_310.java:1285)
	at net.minecraft.class_310.method_1514(class_310.java:882)
	at net.minecraft.client.main.Main.main(Main.java:256)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:243)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:278)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)

-- Renderer details --
Details:
	Assigned renderer: net.minecraft.class_1007@5ff8a237
	Location: 42.53,1.38,-22.36 - World: (42,1,-23), Section: (at 10,1,9 in 2,0,-2; chunk contains blocks 32,-64,-32 to 47,319,-17), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,-64,-512 to 511,319,-1)
	Rotation: -81.5625
	Delta: 0.5795132
Stacktrace:
	at net.minecraft.class_898.method_3954(class_898.java:148)
	at net.minecraft.class_761.method_22977(class_761.java:1312)
	at net.minecraft.class_761.invokeRenderEntity(class_761.java)
	at net.irisshaders.iris.shadows.ShadowRenderer.renderEntities(ShadowRenderer.java:636)
	at net.irisshaders.iris.shadows.ShadowRenderer.renderShadows(ShadowRenderer.java:505)
	at net.irisshaders.iris.pipeline.IrisRenderingPipeline.renderShadows(IrisRenderingPipeline.java:1020)
	at net.minecraft.class_761.handler$zcg000$iris$renderTerrainShadows(class_761.java:5777)
	at net.minecraft.class_761.method_22710(class_761.java:977)
	at net.minecraft.class_757.method_3188(class_757.java:1087)
	at net.minecraft.class_757.method_3192(class_757.java:850)
	at net.minecraft.class_310.method_1523(class_310.java:1285)
	at net.minecraft.class_310.method_1514(class_310.java:882)
	at net.minecraft.client.main.Main.main(Main.java:256)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:243)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:278)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)

-- Uptime --
Details:
	JVM uptime: 2917.401s
	Wall uptime: 2912.796s
	High-res time: 2912.485s
	Client ticks: 58209 ticks / 2910.450s
Stacktrace:
	at net.minecraft.class_310.method_1587(class_310.java:2501)
	at net.minecraft.class_310.method_54580(class_310.java:949)
	at net.minecraft.class_310.method_1514(class_310.java:902)
	at net.minecraft.client.main.Main.main(Main.java:256)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:243)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:278)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)

-- Affected level --
Details:
	All players: 3 total; [class_746['libgcc'/247922, l='ClientLevel', x=583.18, y=93.00, z=771.20], class_745['invertedsunset'/245777, l='ClientLevel', x=583.61, y=78.00, z=770.70], class_745['OnSpray'/263168, l='ClientLevel', x=625.65, y=96.00, z=748.90]]
	Chunk stats: 4096, 705
	Level dimension: minecraft:overworld
	Level spawn location: World: (585,68,756), Section: (at 9,4,4 in 36,4,47; chunk contains blocks 576,-64,752 to 591,319,767), Region: (1,1; contains chunks 32,32 to 63,63, blocks 512,-64,512 to 1023,319,1023)
	Level time: 26729261 game time, 28086322 day time
	Server brand: Paper
	Server type: Non-integrated multiplayer server
	Tracked entity count: 153

-- Last reload --
Details:
	Reload number: 1
	Reload reason: initial
	Finished: Yes
	Packs: vanilla, fabric, fabric-api-base, fabric-block-view-api-v2, fabric-key-binding-api-v1, fabric-rendering-data-attachment-v1, fabric-rendering-fluids-v1, fabric-resource-loader-v0, fabricloader, iris, sodium, file/VanillaTweaks_r510322_MC1.21.x.zip

-- System Details --
Details:
	Minecraft Version: 1.21
	Minecraft Version ID: 1.21
	Operating System: Linux (amd64) version 6.9.6-artix1-1
	Java Version: 22, N/A
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), Oracle Corporation
	Memory: 290408488 bytes (276 MiB) / 680525824 bytes (649 MiB) up to 1073741824 bytes (1024 MiB)
	CPUs: 12
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Ryzen 5 3600 6-Core Processor
	Identifier: AuthenticAMD Family 23 Model 113 Stepping 0
	Microarchitecture: Zen 2
	Frequency (GHz): -0.00
	Number of physical packages: 1
	Number of physical CPUs: 6
	Number of logical CPUs: 12
	Graphics card #0 name: Navi 23 [Radeon RX 6600/6600 XT/6600M]
	Graphics card #0 vendor: Advanced Micro Devices, Inc. [AMD/ATI] (0x1002)
	Graphics card #0 VRAM (MiB): 258.00
	Graphics card #0 deviceId: 0x73ff
	Graphics card #0 versionInfo: unknown
	Virtual memory max (MiB): 16013.33
	Virtual memory used (MiB): 9813.82
	Swap memory total (MiB): 0.00
	Swap memory used (MiB): 0.00
	Space in storage for jna.tmpdir (MiB): <path not set>
	Space in storage for org.lwjgl.system.SharedLibraryExtractPath (MiB): <path not set>
	Space in storage for io.netty.native.workdir (MiB): <path not set>
	Space in storage for java.io.tmpdir (MiB): available: 15977.21, total: 16013.33
	Space in storage for workdir (MiB): available: 209606.52, total: 238419.00
	JVM Flags: 2 total; -Xms512m -Xmx1024m
	Loaded Shaderpack: Sildur's Vibrant Shaders v1.52 High.zip
		Profile: Custom (+1 option changed by user)
	Launched Version: 1.21
	Backend library: LWJGL version 3.3.3-snapshot
	Backend API: AMD Radeon RX 6600 (radeonsi, navi23, LLVM 17.0.6, DRM 3.57, 6.9.6-artix1-1) GL version 4.6 (Core Profile) Mesa 24.1.2-arch1.1, AMD
	Window size: 1920x1080
	GFLW Platform: x11
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: 
	Is Modded: Definitely; Client brand changed to 'fabric'
	Universe: 400921fb54442d18
	Type: Client (map_client.txt)
	Graphics mode: fancy
	Render Distance: 12/12 chunks
	Resource Packs: vanilla, fabric, fabric-api-base, fabric-block-view-api-v2, fabric-key-binding-api-v1, fabric-rendering-data-attachment-v1, fabric-rendering-fluids-v1, fabric-resource-loader-v0, fabricloader, iris, sodium, file/VanillaTweaks_r510322_MC1.21.x.zip
	Current Language: en_us
	Locale: en_US
	System encoding: UTF-8
	File encoding: UTF-8
	CPU: 12x AMD Ryzen 5 3600 6-Core Processor
```

**Comment by Bram1903** [comment]
Same issue for me. Crashes the moment I get too close to another player. Fixed by downgrading to Iris 1.7.1.

**Comment by ATriggeredAmoeba** [comment]
Same issue here

**Comment by ATriggeredAmoeba** [comment]
Same issue here but when I downgrade I get another error that crashes the game when I open the inventory. The issue with the inventory was fixed in 1.7.2 but now I'm getting the same issue with 1.7.2 and cant get into any world. :/

**Comment by vorlitix** [comment]
happens with me as well, tested it with other shaders and it continues to crash.

**Comment by IMS212** [comment]
A fix for this will be released later today.

**Comment by IMS212** [comment]
Fixed in Iris 1.7.3.
